FinniversKit/Sources/Extensions/UIKit/UIControlExtensions.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  Copyright © 2022 FINN AS. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+/// A custom subscription to capture UIControl target events.
+final class UIControlSubscription<SubscriberType: Subscriber, Control: UIControl>: Subscription where SubscriberType.Input == Control {
+    private var subscriber: SubscriberType?
+    private let control: Control
+
+    init(subscriber: SubscriberType, control: Control, event: UIControl.Event) {
+        self.subscriber = subscriber
+        self.control = control
+        control.addTarget(self, action: #selector(eventHandler), for: event)
+    }
+
+    func request(_ demand: Subscribers.Demand) {
+        // We do nothing here as we only want to send events when they occur.
+        // See, for more info: https://developer.apple.com/documentation/combine/subscribers/demand
+    }
+
+    func cancel() {
+        subscriber = nil
+    }
+
+    @objc private func eventHandler() {
+        _ = subscriber?.receive(control)
+    }
+}
+
+/// A custom `Publisher` to work with our custom `UIControlSubscription`.
+struct UIControlPublisher<Control: UIControl>: Publisher {
+
+    typealias Output = Control
+    typealias Failure = Never
+
+    let control: Control
+    let controlEvents: UIControl.Event
+
+    init(control: Control, events: UIControl.Event) {
+        self.control = control
+        self.controlEvents = events
+    }
+    
+    func receive<S>(subscriber: S) where S : Subscriber, S.Failure == UIControlPublisher.Failure, S.Input == UIControlPublisher.Output {
+        let subscription = UIControlSubscription(subscriber: subscriber, control: control, event: controlEvents)
+        subscriber.receive(subscription: subscription)
+    }
+}
+
+/// Extending the `UIControl` types to be able to produce a `UIControl.Event` publisher.
+protocol CombineCompatible { }
+extension UIControl: CombineCompatible { }
+extension CombineCompatible where Self: UIControl {
+    func publisher(for events: UIControl.Event) -> UIControlPublisher<Self> {
+        return UIControlPublisher(control: self, events: events)
+    }
+}
```

Comment:
These extensions could potentially grow long over time, and `CombineCompatible` can be used in combination with other publishers, so maybe we should place these in a separate file?